crates/core/src/syn/parser/kind.rs

```diff
@@ -172,6 +172,7 @@ impl Parser<'_> {
 
 				Ok(Kind::References(table, path))
 			}
+			t!("NONE") => unexpected!(self, next, "a kind name.\nHint: to define a field that can be NONE, use option<type_name> instead."),
```

Comment:
The parsers uses the convention of writing hints within the code block. This can be done with as following:
```suggestion
			t!("NONE") => unexpected!(self, next, "a kind name.", => "to define a field that can be NONE, use option<type_name> instead."),
```
This would result in the following error.
```
error = "There was a problem with the database: Parse error: Unexpected token `NONE`, expected a kind name.
--> [1:41]
  |
1 | DEFINE FIELD name ON user TYPE string | NONE;
  |                                         ^^^^ to define a field that can be NONE, use option<type_name> instead.
```

Reply:
Nice, didn't know the macro had that!